### PR TITLE
Fix a crash when tokenizing a single quote

### LINKED ISSUE #1252: Single quote crashes hy2py
## x.hy

```
'
```

## `hy2py x.hy`

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "hy/cmdline.py", line 406, in hy2py_main
    if stdin_text is None
  File "hy/cmdline.py", line 186, in pretty_error
    print(e, file=sys.stderr)
  File "hy/lex/exceptions.py", line 43, in __str__
    source = self.source.split("\n")
AttributeError: 'NoneType' object has no attribute 'split'
```

### PR DESCRIPTION
**Pull request by Kodiologist**
Fixes #1252.

### CODE REVIEW

File: tests/test_lex.py [hunk 1]
```diff
@@ -63,6 +63,17 @@ def test_unbalanced_exception():
         pass
 
 
+def test_lex_single_quote_err():
+    "Ensure tokenizing \"' \" throws a LexException that can be stringified"
+    # https://github.com/hylang/hy/issues/1252
+    try:
+        tokenize("' ")
+    except LexException as e:
+        assert str(e)
```
Comment (kirbyfan64) [inline tests/test_lex.py]: `== '...'`? To prevent a weird error message coming up without anyone realizing it...

Comment (Kodiologist) [inline tests/test_lex.py]: The string is currently `"\x1b[33m\x1b[22mLexException: Could not identify the next token.\n\n\x1b[39m\x1b[22m"`. If we check for equality against that, we're testing the use of ANSI escape codes, which doesn't seem right.

Comment (kirbyfan64) [inline tests/test_lex.py]: @Kodiologist ...which is why the `in` operator is also equally as awesome as `==`. ;)

Comment (Kodiologist) [inline tests/test_lex.py]: How would `"LexException: Could not identify the next token." in str(e)` tell us whether `str(e)` contains a weird error message, in addition to the one we expect?

Comment (kirbyfan64) [inline tests/test_lex.py]: @Kodiologist Yeah, but it's better than nothing!

Comment (Kodiologist) [inline tests/test_lex.py]: Okay.

**Pull request merged**